src/uct/base/uct_iface.c

```diff
@@ -265,19 +293,21 @@ int uct_iface_is_reachable_v2(const uct_iface_h tl_iface,
 {
     const uct_base_iface_t *iface = ucs_derived_of(tl_iface, uct_base_iface_t);
 
-    if (!ucs_test_all_flags(params->field_mask,
-                            UCT_IFACE_IS_REACHABLE_FIELD_IFACE_ADDR |
-                            UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR)) {
-        ucs_error("missing params (field_mask: %lu), both device_addr and "
-                  "iface_addr should be supplied.", params->field_mask);
-        return 0;
-    }
+    return iface->internal_ops->iface_is_reachable_v2(tl_iface, params);
+}
 
-    if (params->field_mask & UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING) {
-        params->info_string[0] = '\0';
-    }
+int uct_base_iface_is_reachable(const uct_iface_h tl_iface,
+                                const uct_device_addr_t *dev_addr,
+                                const uct_iface_addr_t *iface_addr)
+{
+    uct_iface_is_reachable_params_t params = {
+        .field_mask  = UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR |
+                       UCT_IFACE_IS_REACHABLE_FIELD_IFACE_ADDR,
+        .device_addr = dev_addr,
+        .iface_addr  = iface_addr
+    };
 
-    return iface->internal_ops->iface_is_reachable_v2(tl_iface, params);
+    return uct_iface_is_reachable_v2(tl_iface, &params);
```

Comment:
seems like circular call chain:
uct_base_iface_is_reachable -> uct_iface_is_reachable_v2 -> uct_base_iface_is_reachable_v2 -> uct_iface_is_reachable -> uct_base_iface_is_reachable

Reply:
We now have two base is reachable:
- `uct_base_iface_is_reachable()` -> forwards to v2 `uct_iface_is_reachable_v2()` (`->iface_is_reachable_v2()`)
- `uct_base_iface_is_reachable_v2()` -> eventually forwards to v1 `uct_base_iface_is_reachable()` (`->iface_is_reachable()`)

With this PR, IB is moved to v2. It now uses base v1 to forward to v2. base v2 is not used. Others are not fully moved to v2. They use base v2 to forward to v1. base v2 and base v1 are not used together so we should not have any loop.

Reply:
per offline discussion, will fix in next pr